crates/sui-rest-api/src/checkpoints.rs

```diff
@@ -94,6 +94,106 @@ async fn get_checkpoint_full(
     .pipe(Ok)
 }
 
+pub struct GetCheckpointFullBatch;
+impl ApiEndpoint<RestService> for GetCheckpointFullBatch {
+    fn method(&self) -> axum::http::Method {
+        axum::http::Method::GET
+    }
+
+    fn path(&self) -> &'static str {
+        "/checkpoints/full/batch"
+    }
+
+    fn operation(
+        &self,
+        generator: &mut schemars::gen::SchemaGenerator,
+    ) -> openapiv3::v3_1::Operation {
+        OperationBuilder::new()
+            .tag("Checkpoint")
+            .operation_id("GetCheckpointFullBatch")
+            .query_parameters::<GetCheckpointFullBatchQueryParameters>(generator)
+            .response(
+                200,
+                ResponseBuilder::new()
+                    .json_content::<Vec<CheckpointData>>(generator)
+                    .bcs_content()
+                    .build(),
+            )
+            .response(404, ResponseBuilder::new().build())
+            .response(410, ResponseBuilder::new().build())
+            .build()
+    }
+
+    fn handler(&self) -> RouteHandler<RestService> {
+        RouteHandler::new(self.method(), get_checkpoint_full_batch)
+    }
+}
+
+#[derive(Debug, serde::Serialize, serde::Deserialize, schemars::JsonSchema)]
+pub struct GetCheckpointFullBatchQueryParameters {
+    pub start: CheckpointSequenceNumber,
+    pub batch_size: u32,
```

Comment:
Look at the other list apis (checkpoints as an example) to match the param naming